tools/distpartitioning/globalids.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+import torch
+import operator
+import itertools
+from gloo_wrapper import allgather_sizes, alltoall_cpu, alltoallv_cpu
+
+def get_shuffle_global_nids(rank, world_size, global_nids_ranks, parts_map, node_data):
+    """ For nodes which are not owned by the current rank, whose global_nid <-> shuffle_global-nid 
+    is not present at the current rank, this function retrieves their shuffle_global_ids from the owning
+    rank
+
+    Parameters: 
+    -----------
+    rank : integer
+        rank of the process
+    world_size : integer
+        total no. of ranks configured
+    global_nids_ranks : list
+        list of lists of global_nids, index of the list is the rank of the process
+                    where global_nid <-> shuffle_global_nid mapping is located. 
```

Comment:
are the data still stored in a list of list? or a list of tensors?

Reply:
Now data is stored in list of numpy arrays.